ui/src/components/Client.js

```diff
@@ -4,26 +4,38 @@ import ConfigurationView from './configuration/ConfigurationView';
 import MobileClientBuildsList from './build/MobileClientBuildsList';
 
 const listBuildsUrl = `/api/builds`;
+const listBuildConfigsUrl = '/api/buildconfigs';
 
 class Client extends Component {
 
   constructor(props) {
     super(props);
 
     this.state = {
-      mobileClientBuilds: []
+      buildConfigs: []
     };
   }
 
-  componentDidMount = () => {
-    fetch(listBuildsUrl, {credentials: "same-origin"})
-      .then(response => response.json())
-      .then(result => {
-        this.setState({mobileClientBuilds: result.items});
-      })
-      .catch(err => {
-        console.error('Fetch error: ', err)
+  componentDidMount = async () => {
+    try {
+      let configs = await fetch(listBuildConfigsUrl, { credentials: 'same-origin' });
```

Comment:
I think there is a `DataService` module has all the code that need to interact with the back end. Can we move this to there as well?